server/src/test/kotlin/org/itmodreamteam/myrest/server/service/restaurant/RegisterTest.kt

```diff
@@ -0,0 +1,92 @@
+package org.itmodreamteam.myrest.server.service.restaurant
+
+import org.assertj.core.api.Assertions.assertThat
+import org.itmodreamteam.myrest.server.error.UserException
+import org.itmodreamteam.myrest.server.model.restaurant.Restaurant
+import org.itmodreamteam.myrest.server.repository.restaurant.RestaurantRepository
+import org.itmodreamteam.myrest.shared.restaurant.RestaurantRegistrationInfo
+import org.itmodreamteam.myrest.shared.restaurant.RestaurantStatus
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+import org.springframework.boot.test.context.TestConfiguration
+import org.springframework.context.annotation.ComponentScan
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.context.junit4.SpringRunner
+import javax.validation.ConstraintViolationException
+
+@RunWith(SpringRunner::class)
+@DataJpaTest
+@ContextConfiguration(classes = [RegisterTest.Config::class])
+class RegisterTest {
+
+    @Autowired
+    lateinit var restaurantService: RestaurantService
+
+    @Autowired
+    lateinit var restaurantRepository: RestaurantRepository
+
+    private lateinit var restaurant: Restaurant
+
+    @Before
+    fun setup() {
+        restaurant = restaurantRepository.save(Restaurant("Pizza", "Italian food", "ИНН"))
+    }
+
+    @Test(expected = UserException::class)
+    fun `Given existing restaurant, when create restaurant with existing name, then failure`() {
+        restaurantService.register(RestaurantRegistrationInfo("Pizza", "", ""))
+    }
+
+    @Test
+    fun `When create new restaurant with valid required arguments, then new restaurant created`() {
+        restaurantService.register(RestaurantRegistrationInfo("Pasta", "Italian food", "INN"))
+
+        val restaurants = restaurantRepository.findAll()
+        assertThat(restaurants).hasSize(2)
+
+        val newRestaurant = restaurants[1]
+        assertThat(newRestaurant.name).isEqualTo("Pasta")
+        assertThat(newRestaurant.description).isEqualTo("Italian food")
+        assertThat(newRestaurant.legalInfo).isEqualTo("INN")
+        assertThat(newRestaurant.status).isEqualTo(RestaurantStatus.PENDING)
+    }
+
+    @Test
+    fun `When create new restaurant with the whole arguments, then new restaurant created`() {
+        var createdRestaurant = RestaurantRegistrationInfo(
+            "Pasta",
+            "Italian food",
+            "docs",
+            "pizzapasta.com",
+            "8928335050",
+            "pizza@mail.ru"
+        )
+        restaurantService.register(createdRestaurant)
+
+        val restaurants = restaurantRepository.findAll()
```

Comment:
Convenience: if method register() returned ID, you would be able to fetch the entity by it. Here you extract the second restaurant in a list, but who guarantees it's the newly created one?